transports/tcp/src/lib.rs

```diff
@@ -927,7 +938,16 @@ mod tests {
             let mut listener = tcp.clone().listen_on(addr).unwrap();
             match listener.next().await.unwrap().unwrap() {
                 ListenerEvent::NewAddress(_) => {
+                    // Check that tcp and listener share the same port reuse SocketAddr
+                    let port_reuse_tcp = tcp.port_reuse.local_dial_addr(&listener.listen_addr.ip());
+                    let port_reuse_listener = listener
+                        .port_reuse
+                        .local_dial_addr(&listener.listen_addr.ip());
+                    assert!(port_reuse_tcp.is_some());
+                    assert_eq!(port_reuse_tcp, port_reuse_listener);
+
                     // Obtain a future socket through dialing
+                    // FIXME: Check that the port used by socket is the same as the port returned by local_dial_addr
```

Comment:
How about checking the `ListenerEvent::Upgrade { remote_addr, ..}` in `async fn listener` above instead?

Reply:
OK I updated the code using an mpsc channel to pass the reused port between the dialer and the listener, and cheking in the listener upgrade event that the remote port is the same as the one received in the channel.